docs/pages/world/modules/keysintable.mdx

```diff
@@ -0,0 +1,239 @@
+import { Callout } from "nextra/components";
+
+import { CollapseCode } from "../../../components/CollapseCode";
+
+# Keys in Table Module
+
+[The `KeysInTable` module](https://github.com/latticexyz/mud/tree/main/packages/world-modules/src/modules/keysintable) tracks which keys are used with a specific table.
+This is useful if you need to be able to iterate through all the records of a table onchain.
+
+<Callout type="warning" emoji="💰">
+It costs additional gas for every write to the table affected.
+If you just need this functionality offchain it is best to look at the data offchain instead.
+
+Also, this is a _generic_ solution.
+You can usually save gas by implementing a solution that is specific to your use case, using [hooks](/store/store-hooks).
```

Comment:
its the hooks that are the costly pattern, its better to implement this manually and use-case-specific to save gas

Reply:
can we remove the suggestion for hooks since that defeats the purpose of the custom solution